tests/nuodb_cursor_test.py

```diff
@@ -125,6 +127,29 @@ def test_executemany_somefail(self):
 
         cursor.execute("DROP TABLE executemany_table")
 
+    def test_result_set_gets_closed(self):
+        current_version = getenv('NUODB_VERSION', None) # skipif <2.1
+        if current_version is not None and not LooseVersion(current_version) < LooseVersion("2.1"):
+        # Server will throw error after 1000 open result sets
+            con = self._connect()
+            for j in [False, True]:
```

Comment:
Seems like this would be more straightforward in two separate loops. The first loop successfully does 2015 (?) queries. The second loop expects a failure on the 1000th query. Is that correct?
Again, you said this was a regression - so I'm wondering how this used to work. Was it the case that previously you could execute more than 1000 queries without having to call close on the cursor?